autopilot/autopilot.go

```diff
@@ -249,7 +274,10 @@ func (ap *Autopilot) Run() error {
 			}
 
 			// update current period
-			ap.c.updateCurrentPeriod()
+			err = ap.c.updateCurrentPeriod(ctx)
+			if err != nil {
+				ap.logger.Warnf("failed to update current period, err: %v", err)
```

Comment:
If this fails we probably won't be able to do anything else either. Might be worthwhile to just skip this iteration then.